### ISSUE TITLE
WebDriverManager fails to setup when using junit.jupiter.execution.parallel.enabled true 

### ISSUE THREAD

**Issue by tyge68** [bonigarcia/selenium-jupiter#80]
**Description of the problem**: When using parallelism of Junit5 feature, on our Linux jenkins slave (which are always cleaned up, mvn repository is usually cleaned up). Tests are failing while trying to execute WebDriverManager.setup(..) sub routines. This never occurs if we run tests sequentially. So something is not "concurrent" friendly in this code, maybe it should be "synchronized" to be more safe ?!?

**Browser and version**: chrome-in-docker
**Operating system**:  Linux 64bits.

**Selenium-Jupiter version**:  3.3.4
**Selenium-Jupiter use**:  Nothing specific even empty test that request a driver would do.

**Selenium-Jupiter traces**: 
11:10:38  09:10:38.222 [ForkJoinPool-1-worker-3] ERROR i.g.b.w.WebDriverManager - There was an error managing chromedriver 83.0.4103.39 (Parameter 'directory' is not a directory: /home/****/.m2/repository/webdriver)
11:10:38  java.lang.IllegalArgumentException: Parameter 'directory' is not a directory: /home/****/.m2/repository/webdriver
11:10:38  	at org.apache.commons.io.FileUtils.validateListFilesParameters(FileUtils.java:536)
11:10:38  	at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:512)
11:10:38  	at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:684)
11:10:38  	at io.github.bonigarcia.wdm.WebDriverManager.getFilesInCache(WebDriverManager.java:926)
11:10:38  	at io.github.bonigarcia.wdm.WebDriverManager.getDriverFromCache(WebDriverManager.java:869)
11:10:38  	at io.github.bonigarcia.wdm.WebDriverManager.handleCache(WebDriverManager.java:860)
11:10:38  	at io.github.bonigarcia.wdm.WebDriverManager.downloadAndExport(WebDriverManager.java:557)
11:10:38  	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:547)
11:10:38  	at io.github.bonigarcia.wdm.WebDriverManager.handleException(WebDriverManager.java:773)
11:10:38  	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:550)
11:10:38  	at io.github.bonigarcia.wdm.WebDriverManager.handleException(WebDriverManager.java:766)
11:10:38  	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:550)
11:10:38  	at io.github.bonigarcia.wdm.WebDriverManager.setup(WebDriverManager.java:260)
11:10:38  	at io.github.bonigarcia.seljup.SeleniumExtension.runWebDriverManagerIfNeded(SeleniumExtension.java:262)
11:10:38  	at io.github.bonigarcia.seljup.SeleniumExtension.resolveParameter(SeleniumExtension.java:217)
11:10:38  	at io.github.bonigarcia.seljup.handler.RemoteDriverHandler.resolveGeneric(RemoteDriverHandler.java:194)
11:10:38  	at io.github.bonigarcia.seljup.handler.RemoteDriverHandler.resolveOtherThanDocker(RemoteDriverHandler.java:133)
11:10:38  	at io.github.bonigarcia.seljup.handler.RemoteDriverHandler.resolve(RemoteDriverHandler.java:106)
11:10:38  	at io.github.bonigarcia.seljup.SeleniumExtension.resolveHandler(SeleniumExtension.java:309)
11:10:38  	at io.github.bonigarcia.seljup.SeleniumExtension.resolveParameter(SeleniumExtension.java:222)
11:10:38  	at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameter(ExecutableInvoker.java:216)
11:10:38  	at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameters(ExecutableInvoker.java:183)
11:10:38  	at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameters(ExecutableInvoker.java:144)
11:10:38  	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:96)
11:10:38  	at

**Comment by tyge68** [comment]
I think it's not clear why actually it tries to use / download webdriver as in this context it uses a chrome in docker, maybe it couldn't start the docker in time.

I am trying to tune some parameters (i.e sel.jup.docker.wait.timeout.sec and sel.jup.remote.webdriver.wait.timeout.sec) it would be nice also to be able to completely disable the fallback mechanism then. So we could definitely see in the stack trace the root cause of the failure I guess.

**Comment by bonigarcia** [comment]
I suppose there is some kind of race condition here. But I agree that WebDriverManager should not be called when using browsers in Docker container. Could you please try to collect the Selenium-Jupiter traces and paste them here?

**Comment by tyge68** [comment]
I have uploaded the trace log, maybe you can find the root cause why it fails on this environment to run in parallel.
On my local  (Mac OS) it doesn't have the issue, so maybe it related to Linux / Docker.

[trace_log.txt](https://github.com/bonigarcia/selenium-jupiter/files/5207336/trace_log.txt)

**Comment by MichielBugherJelli** [comment]
@bonigarcia any updates here?
If you're busy and need help investigating, any guidance or pointing in a direction would help.

**Comment by bonigarcia** [comment]
Sorry for the late reply. I'm working in a new major release of Selenium-Jupiter currently. In the new release, the capabilities to create Docker containers is delegated on WebDriverManager. Therefore, this problem might happen or not in the new release. I will inform you when this new release is available.

**Comment by bonigarcia** [comment]
Selenium-Jupiter 4 is out. Please have a try with that version. If the error still persists, please open another issue.

**Comment by tyge68** [comment]
@bonigarcia currently I cannot update to latest version as we also use selenide, which is not compatible with latest selenium version used -> 4.0.0-beta-4, it still depends on 3.141.59 (1) , which then fails to load the chromiumdriver which is new in 4.0.0-beta-4

(1) https://github.com/selenide/selenide/blob/master/gradle/dependencies.gradle#L8
